snuba/query/processors/physical.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+
+from snuba.query.physical import Query
+from snuba.request.request_settings import RequestSettings
+
+
+class QueryProcessor(ABC):
+    """
+    This class represents the same concept as the Logical QueryProcessor (see
+    query.processors.logical).
+    The only difference is that it works on the Physical Query instead of the
+    Logical Query.
```

Comment:
I was thinking about whether to do that, I did not because I could not find an actual use for the generic. These are two disjoint classes (logical and physical) that just happen to have the same interface conceptually. But they are never used interchangeably (that would be really wrong), and they are fully abstract so they do not have any parametric behavior.
I am open to adding the generic, what do you think ?

Reply:
If it is more coincidental than intentional, I agree that it probably doesn't make much sense to make it generic. At the same time, I'd probably avoid implying any intentional relationship to the logical query in the docstring as well (rather than explaining its behavior in relation to a different class, it should probably just describe it on its own terms.) It might also avoid any confusion down the line if the logical query processor changes for some reason.

Reply:
That makes sense, will rewrite the docstring

Reply:
Docstring is rewritten now.